### PR TITLE
fix: Set g.user to anon user in Celery

### PR DESCRIPTION
**Pull request by benjreinhart**
### SUMMARY

**TLDR:** this change should enable async queries for public/anonymous users.

The async queries experience runs queries in Celery workers instead of during the web request. Queries check some state against the requesting user (e.g., `can_access`) and assumes the presence of a `flask.g.user` object. The Celery workers therefore need to set the user on `flask.g` which today they do for requests that were initiated by a logged in user. However, it is broken for anonymous users since there is no user id to pass to the Celery workers.

### TESTING INSTRUCTIONS

Manual

### Notes

I didn't seem to find a cleaner way to grab a reference to an anonymous user. Interested to know if there's a better approach than reaching into flask login for this. @dpgaspar maybe?

cc @robdiciuccio @dpgaspar

### CODE REVIEW

**Review by robdiciuccio — COMMENTED** [review]
Seems reasonable to me, but I'd like to get @dpgaspar 's opinion.

File: superset/tasks/async_queries.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def ensure_user_is_set(user_id: Optional[int]) -> None:
     user_is_set = hasattr(g, "user") and g.user is not None
     if not user_is_set and user_id is not None:
         g.user = security_manager.get_user_by_id(user_id)
+    else:
+        g.user = security_manager.get_anonymous_user()
```
Comment (villebro) [inline superset/tasks/async_queries.py:47]: Wouldn't this set `g.user` to anonymous is `user_is_set`? It appears we should make this `elif not user_is_set:`?

File: tests/tasks/async_queries_tests.py [hunk 2]
```diff
@@ -202,3 +203,29 @@ def test_soft_timeout_load_explore_json_into_cache(
...
+        self.assertFalse(hasattr(g, "user"))
+        ensure_user_is_set(None)
+        self.assertTrue(hasattr(g, "user"))
+        self.assertTrue(g.user.is_anonymous)
+        self.assertEqual(None, g.user.get_id())
+
+        if g_user_is_set:
+            g.user = original_g_user
+        else:
+            del g.user
```
Comment (villebro) [inline tests/tasks/async_queries_tests.py:231]: In reference to the above comment, should we add a test case for cases where `g.user` is set and see what `g.user` is after calling `ensure_user_is_set`?

**Review by villebro — COMMENTED** [review]
One observation (I'm no expert on flask globals, so forgive me if I'm in the wrong here!)

File: superset/tasks/async_queries.py [hunk 3]
```diff
@@ -43,6 +43,8 @@ def ensure_user_is_set(user_id: Optional[int]) -> None:
     user_is_set = hasattr(g, "user") and g.user is not None
```
Comment (dpgaspar) [inline superset/tasks/async_queries.py:43]: Good candidate for security manager

File: superset/tasks/async_queries.py (hunk 1 above)
Comment (benjreinhart) [reply superset/tasks/async_queries.py]: You are right, good catch!

**Review by robdiciuccio — APPROVED** [review]
